File: src/main/java/id/putra/wms/module/supplier/model/entity/Supplier.java
package id.putra.wms.module.supplier.model.entity;

import java.io.Serializable;
import java.util.List;

import id.putra.wms.shared.base.entity.BaseEntity;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Table
@Entity
@Data
@EqualsAndHashCode(callSuper = false)
public class Supplier extends BaseEntity implements Serializable {
    @Id
    private String id;

    private String code;

    private String name;

    private Boolean isActive;

    @OneToMany(mappedBy = "supplier", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ProductSupplier> products;

    @OneToMany(mappedBy = "supplier", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ContactPersonSupplier> contactPersonSuppliers;
}

File: web/types/supplier.ts
import type { Product } from "./product";

export type ContactPersonSupplier = {
  id: number | null;
  name: string;
  email: string;
  phone: string;
  address: string;
};

export type ProductSupplierId = {
  product_id: number;
  supplier_id: number | null;
};

export type ProductSupplier = {
  id: ProductSupplierId;
  product: Product;
};

export type Supplier = {
  id: number | null;
  name: string;
  code: string;
  is_active: boolean;
  contact_person_suppliers: ContactPersonSupplier[];
  products: ProductSupplier[];
};
